pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -36,99 +40,92 @@ var (
 	//go:embed yaml/deployment-testenv-p.yaml
 	testenvDeploymentP string
 
-	InjectedAnnotationsSMCPPatch = `
-spec:
-  proxy:
-    injection:
-      autoInject: true
-      injectedAnnotations:
-        test1.annotation-from-smcp: test1
-        test2.annotation-from-smcp: '["test2"]'
-        test3.annotation-from-smcp: '{test3}'
-`
+	InjectedAnnotationsSMCPPatch = `{"spec":{"proxy":{"injection":{"autoInject":true,"injectedAnnotations":{"test1.annotation-from-smcp":"test1","test2.annotation-from-smcp":"[\"test2\"]","test3.annotation-from-smcp":"{test3}"}}}}}`
 )
 
-func cleanupSMCPAnnotations() {
-	log.Log.Info("Cleanup ...")
-	util.KubeDeleteContents("bookinfo", testenvDeploymentX86)
-	time.Sleep(time.Duration(20) * time.Second)
-}
-
 func TestSMCPAnnotations(t *testing.T) {
-	test.NewTest(t).Id("T29").Groups(test.Full).NotRefactoredYet()
-
-	defer cleanupSMCPAnnotations()
-
-	t.Run("smcp_test_annotation_proxyEnv", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Test annotation sidecar.maistra.io/proxyEnv")
-		if env.Getenv("SAMPLEARCH", "x86") == "p" {
-			util.KubeApplyContents("bookinfo", testenvDeploymentP)
-		} else if env.Getenv("SAMPLEARCH", "x86") == "z" {
-			util.KubeApplyContents("bookinfo", testenvDeploymentZ)
-		} else {
-			util.KubeApplyContents("bookinfo", testenvDeploymentX86)
-		}
-		util.CheckPodRunning("bookinfo", "app=env")
-		msg, err := util.ShellMuteOutput(`kubectl get po -n bookinfo -o yaml | grep maistra_test_env`)
-		util.Inspect(err, "Failed to get variables", "", t)
-
-		if strings.Contains(msg, "env_value") {
-			log.Log.Info(msg)
-		} else {
-			t.Errorf("Failed to get env variable: %v", msg)
-		}
-		util.KubeDeleteContents("bookinfo", testenvDeploymentX86)
-		time.Sleep(time.Duration(30) * time.Second)
+	NewTest(t).Id("T29").Groups(Full).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		t.NewSubTest("proxyEnvoy").Run(func(t TestHelper) {
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, ns)
+			})
+			t.LogStep("Deploy TestSSL pod with annotations sidecar.maistra.io/proxyEnv")
+			DeployTestSsl(t, ns)
+
+			t.LogStep("Get annotations and verify that the pod has the expected ones")
+			annotations := make(map[string]string)
+			annotations["sidecar.maistra.io/proxyEnv"] = `{ "maistra_test_env": "env_value", "maistra_test_env_2": "env_value_2" }`
```

Comment:
This can be simplified:
```suggestion
			annotations := map[string]string {
				"sidecar.maistra.io/proxyEnv": `{ "maistra_test_env": "env_value", "maistra_test_env_2": "env_value_2" }`
			}
```